jsons/_date_impl.py

```diff
@@ -0,0 +1,28 @@
+"""
```

Comment:
With only some minor changes, you could fit this functionality in `_datetime_impl.py` and saving a new module:

```python
RFC3339_DATE_PATTERN = '%Y-%m-%d'
RFC3339_TIME_PATTERN = '%H:%M:%S'
RFC3339_DATETIME_PATTERN = '{}T{}'.format(
    RFC3339_DATE_PATTERN, RFC3339_TIME_PATTERN)


def to_str(
        dt: Union[datetime, date],
        strip_microseconds: bool,
        fork_inst: type,
        pattern: str = RFC3339_DATETIME_PATTERN) -> str:
    offset = get_offset_str(dt, fork_inst)
    if not strip_microseconds and getattr(dt, 'microsecond', None):
        pattern += '.%f'
    return dt.strftime("{}{}".format(pattern, offset))


def get_offset_str(
        obj: Union[datetime, date, timedelta],
        fork_inst: type) -> str:
    """
    Return the textual offset of the given ``obj``.
    :param obj: a datetime or timedelta instance.
    :param fork_inst: the state holder that is used.
    :return: the offset following RFC3339.
    """
    result = ''
    if isinstance(obj, datetime):
        result = _datetime_offset_str(obj, fork_inst)
    elif isinstance(obj, timedelta):
        result = _timedelta_offset_str(obj, fork_inst)
    return result


def get_datetime_inst(obj: str, pattern: str) -> datetime:
    """
    Return a datetime instance with timezone info from the given ``obj``.
    :param obj: the ``obj`` in RFC3339 format.
    :param pattern: the datetime pattern.
    :return: a datetime instance with timezone info.
    """
    if obj[-1] == 'Z':
        result = _datetime_utc_inst(obj, pattern)
    elif 'T' in pattern:
        result = _datetime_offset_inst(obj, pattern)
    else:
        result = datetime.strptime(obj, pattern)
    return result
```

This is how you can use it:

```python
d_in = datetime.now().date()
d_str = to_str(d_in, False, StateHolder, pattern=RFC3339_DATE_PATTERN)
d_out = get_datetime_inst(d_str, RFC3339_DATE_PATTERN)
```